packages/web/src/pages/developer-page/DeveloperPage.tsx

```diff
@@ -0,0 +1,140 @@
+const createDeveloperApp = async ({
+  userId,
+  name,
+  isPersonalAccess
+}: CreateDeveloperAppRequest) => {
+  const sdk = await audiusSdk()
+  const res = await sdk.developerApps.createDeveloperApp({
+    userId,
+    name,
+    isPersonalAccess
+  })
+  return { apiKey: res.apiKey, apiSecret: res.apiSecret }
+}
+
+// TODO: Move to saga, add confirmation (unless we confirm in SDK itself)
+const deleteDeveloperApp = async ({
+  userId,
+  appApiKey
+}: DeleteDeveloperAppRequest) => {
+  const sdk = await audiusSdk()
+  await sdk.developerApps.deleteDeveloperApp({
+    userId,
+    appApiKey
+  })
+  return appApiKey
+}
+
+export const DeveloperPage = () => {
+  const [createAppResult, setCreateAppResult] = useState<null | {
+    apiKey: string
+    apiSecret: string
+  }>(null)
+  const [deleteSuccess, setDeleteSuccess] = useState<boolean | null>(null)
+  const currentUserId = useSelector((state) => {
+    return getAccountUser(state)?.user_id
+  })
+
+  const reset = () => {
+    setCreateAppResult(null)
+    setDeleteSuccess(null)
+  }
+
+  const handleSubmitDeveloperApp = useCallback(
+    async (isPersonalAccess: boolean) => {
+      if (!currentUserId) return
+      reset()
+      let res: { apiKey: string; apiSecret: string } | null = null
+      const encodedUserId = encodeHashId(currentUserId)
+      try {
+        res = await createDeveloperApp({
+          userId: encodedUserId,
+          name: 'Test app',
+          isPersonalAccess
+        })
+      } catch (e) {
+        console.error('There was an error: ', e)
+        return
+      }
+
+      // TODO: Ideally move this logic to SDK's createDeveloperApp
+      if (isPersonalAccess) {
+        try {
+          await createGrant({ appApiKey: res.apiKey, userId: encodedUserId })
+        } catch (e) {
+          console.error('There was an error: ', e)
+        }
+      }
+      setCreateAppResult(res)
+    },
+    [currentUserId]
+  )
+
+  const handleDeleteDeveloperApp = useCallback(async () => {
+    if (!currentUserId || !createAppResult) return
+    try {
+      await deleteDeveloperApp({
+        userId: encodeHashId(currentUserId),
+        appApiKey: createAppResult.apiKey
+      })
+    } catch (e) {
+      console.error('There was an error: ', e)
+    }
+    setDeleteSuccess(true)
+  }, [createAppResult, currentUserId])
+
+  return (
+    <div className={styles.container}>
+      <button onClick={() => handleSubmitDeveloperApp(false)}>
+        Create Developer App
+      </button>
+      <button onClick={() => handleSubmitDeveloperApp(true)}>
+        Create Personal Use API Key
+      </button>
+      <div>
+        {createAppResult == null ? null : (
+          <div>
+            <div>
+              <p>Success! Here are your app&apos;s credentials:</p>
+              <p>
+                API Key: {createAppResult.apiKey}
+                <br />
+                API Secret: {createAppResult.apiSecret}
```

Comment:
I assume we'll have design at some point around this but thought I'd mention anyways that this probably makes sense to put in a password field

Reply:
oh yes 100%